### DISCUSSION TITLE
Transpiling any language to JS and the security model

### DISCUSSION THREAD

**Discussion by PawelStadnicki** [observablehq/framework#1875] [Q&A]
I am not a professional Node.js developer,
it may be a silly question for something that works by design.

This question has origin in [open code tag up for extension ](https://github.com/observablehq/framework/issues/967)

I read the [docs](https://observablehq.com/documentation/security/security-and-data-access), but I'm not sure if/how the backend transpilation fits there.
How does the Framework ensure integrity in the case of backend requests for live coding?

In particular, let’s assume I have code in any language that compiles to JavaScript on the backend:

let evaluate value = 1.67 // any value or computation

This code is sent to the backend for transpilation, and JavaScript code is returned that can be invoked in the browser.
How do we ensure that my evaluate function is not mixed up with the requests of other simultaneous users of the dashboard?

Similarly, if I have a function like:

let continuation x = evaluate x // evaluate definition may vary between users

Are the definitions of my functions always isolated from other app users?
Is this something that Node.js ensures by design, or does the Framework handle this isolation explicitly?

Many thanks in advance!

**Comment by Fil** [comment]
Only the code that is in your app’s `src` directory is transpiled. In a sense, you can't have multiple users (or maybe it's a shared directory to which different users have access, but that is beyond the control of Framework).
